<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>SPEAQeasy: a Scalable Pipeline for Expression Analysis and Quantification for R/Bioconductor-powered RNA-seq analyses • SPEAQeasyWorkshop2020</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="SPEAQeasy: a Scalable Pipeline for Expression Analysis and Quantification for R/Bioconductor-powered RNA-seq analyses">
<meta property="og:description" content="SPEAQeasyWorkshop2020">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SPEAQeasyWorkshop2020</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/SPEAQeasyWorkshop2020.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="SPEAQeasyWorkshop2020_files/accessible-code-block-0.0.1/empty-anchor.js"></script><link href="SPEAQeasyWorkshop2020_files/anchor-sections-1.0/anchor-sections.css" rel="stylesheet">
<script src="SPEAQeasyWorkshop2020_files/anchor-sections-1.0/anchor-sections.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>SPEAQeasy: a Scalable Pipeline for Expression Analysis and Quantification for R/Bioconductor-powered RNA-seq analyses</h1>
                        <h4 class="author">Nicholas J. Eagles</h4>
            <address class="author_afil">
      Lieber Institute for Brain Development, Johns Hopkins Medical Campus<br><h4 class="author">Joshua M. Stolz</h4>
            <address class="author_afil">
      Lieber Institute for Brain Development, Johns Hopkins Medical Campus<br><h4 class="author">Leonardo Collado-Torres</h4>
            <address class="author_afil">
      Lieber Institute for Brain Development, Johns Hopkins Medical Campus<br><a class="author_email" href="mailto:#"></a><a href="mailto:lcolladotor@gmail.com" class="email">lcolladotor@gmail.com</a>
      </address>
                  
            <h4 class="date">15 December 2020</h4>
      
      
      <div class="hidden name"><code>SPEAQeasyWorkshop2020.Rmd</code></div>

    </address>
</address>
</div>

    
    
<div id="overview" class="section level1">
<h1 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h1>
<p>This workshop aims to describe the <code>SPEAQeasy</code> (Eagles, Burke, Leonard, Barry, Stolz, Huuki, Phan, Serrato, Gutiérrez-Millán, Aguilar-Ordoñez, Jaffe, and Collado-Torres, 2020) RNA-seq processing pipeline, show how to use it, and then illustrate how the results can be analyzed using Bioconductor R packages for differential expression analyses.</p>
<p>SPEAQeasy is a <a href="https://www.nextflow.io/">Nextflow</a>-based <strong>S</strong>calable RNA-seq processing <strong>P</strong>ipeline for <strong>E</strong>xpression <strong>A</strong>nalysis and <strong>Q</strong>uantification that produces R objects ready for analysis with Bioconductor tools. Partipants will become familiar with SPEAQeasy set-up, execution on real data, and practice configuring some common settings. We will walk through a complete differential expression analysis, utilizing popular packages such as <a href="https://www.bioconductor.org/packages/release/bioc/html/limma.html">limma</a>, <a href="http://bioconductor.org/packages/release/bioc/html/edgeR.html">edgeR</a>, and <a href="https://bioconductor.org/packages/release/bioc/html/clusterProfiler.html">clusterProfiler</a>.</p>
<div id="instructors" class="section level2">
<h2 class="hasAnchor">
<a href="#instructors" class="anchor"></a>Instructors</h2>
<ul>
<li><a href="https://github.com/Nick-Eagles">Nick Eagles</a></li>
<li><a href="https://github.com/joshstolz">Josh Stolz</a></li>
</ul>
</div>
<div id="workshop-description" class="section level2">
<h2 class="hasAnchor">
<a href="#workshop-description" class="anchor"></a>Workshop Description</h2>
<p>TODO</p>
</div>
<div id="pre-requisites" class="section level2">
<h2 class="hasAnchor">
<a href="#pre-requisites" class="anchor"></a>Pre-requisites</h2>
<ul>
<li>Basic understanding of RNA-seq</li>
<li>Basic familiarity with the <code>SummarizedExperiment</code> and <code>GenomicRanges</code> packages</li>
</ul>
</div>
<div id="workshop-participation" class="section level2">
<h2 class="hasAnchor">
<a href="#workshop-participation" class="anchor"></a>Workshop Participation</h2>
<p>This workshop makes use of two sets of files: the <a href="https://github.com/LieberInstitute/SPEAQeasy">SPEAQeasy GitHub repository</a> and the workshop-specific files. You can clone the SPEAQeasy repository with:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1"></a><span class="fu">git</span> clone git@github.com:LieberInstitute/SPEAQeasy.git</span></code></pre></div>
<p>The workshop-specific files include example outputs from SPEAQeasy, for performing a differential expression analysis. You can download a Docker image with these files using:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1"></a><span class="ex">docker</span> run -e PASSWORD=bioc2020 -p 8787:8787 -d --rm lcollado/speaqeasyworkshop2020</span></code></pre></div>
<p>Then, log in to RStudio at <a href="http://localhost:8787">http://localhost:8787</a> using username <code>rstudio</code> and password <code>bioc2020</code>. Note that on Windows you need to provide your localhost IP address like <code>http://191.163.92.108:8787/</code> - find it using <code>docker-machine ip default</code> in Docker’s terminal.</p>
<p>To see the vignette on RStudio’s window (from the docker image), run <code><a href="https://rdrr.io/r/utils/browseVignettes.html">browseVignettes(package = "SPEAQeasyWorkshop2020")</a></code>. Click on one of the links, “HTML”, “source”, “R code”. In case of <code>The requested page was not found</code> error, add <code>help/</code> to the URL right after the hostname, e.g., <a href="http://localhost:8787/help/library/SPEAQeasyWorkshop2020/doc/SPEAQeasyWorkshop2020.html">http://localhost:8787/help/library/SPEAQeasyWorkshop2020/doc/SPEAQeasyWorkshop2020.html</a>. This is a <a href="https://github.com/rocker-org/rocker-versioned/issues/178">known bug</a>.</p>
</div>
<div id="time-outline" class="section level2">
<h2 class="hasAnchor">
<a href="#time-outline" class="anchor"></a>Time outline</h2>
<table class="table">
<thead><tr class="header">
<th>Activity</th>
<th>Time</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>General overview of SPEAQeasy</td>
<td>20m</td>
</tr>
<tr class="even">
<td>Hands-on: configuring and running SPEAQeasy on real data</td>
<td>25m</td>
</tr>
<tr class="odd">
<td>Understanding SPEAQeasy outputs</td>
<td>15m</td>
</tr>
<tr class="even">
<td>Differential expression analysis</td>
<td>30m</td>
</tr>
</tbody>
</table>
<p>Total: a 90 minute session.</p>
</div>
<div id="workshop-goals-and-objects" class="section level2">
<h2 class="hasAnchor">
<a href="#workshop-goals-and-objects" class="anchor"></a>Workshop goals and objects</h2>
<div id="learning-goals" class="section level3">
<h3 class="hasAnchor">
<a href="#learning-goals" class="anchor"></a>Learning goals</h3>
<ul>
<li>Understand what SPEAQeasy is and how it can fit into a complete RNA-seq processing workflow</li>
<li>Become familiar with running SPEAQeasy on real input data</li>
<li>Understand SPEAQeasy outputs and the Bioconductor packages available for different analysis goals</li>
<li>Get concrete experience with an example differential expression analysis given SPEAQeasy output data</li>
</ul>
</div>
<div id="learning-objectives" class="section level3">
<h3 class="hasAnchor">
<a href="#learning-objectives" class="anchor"></a>Learning objectives</h3>
</div>
</div>
<div id="citing-speaqeasy" class="section level2">
<h2 class="hasAnchor">
<a href="#citing-speaqeasy" class="anchor"></a>Citing <code>SPEAQeasy</code>
</h2>
<p>We hope that <code>SPEAQeasy</code> (Eagles, Burke, Leonard, et al., 2020) will be useful for your research. Please use the following information to cite the package and the overall approach. Thank you!</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Citation info</span>
<span class="fu"><a href="https://rdrr.io/r/utils/citation.html">citation</a></span><span class="op">(</span><span class="st">"SPEAQeasyWorkshop2020"</span><span class="op">)</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Eagles NJ, Burke EE, Leonard J, Barry BK, Stolz JM, Huuki L, Phan BN,</span>
<span class="co">#&gt; Serrato VL, Gutiérrez-Millán E, Aguilar-Ordoñez I, Jaffe AE,</span>
<span class="co">#&gt; Collado-Torres L (2020). "SPEAQeasy: a Scalable Pipeline for Expression</span>
<span class="co">#&gt; Analysis and Quantification for R/Bioconductor-powered RNA-seq</span>
<span class="co">#&gt; analyses." _bioRxiv_. doi: 10.1101/2020.12.11.386789 (URL:</span>
<span class="co">#&gt; https://doi.org/10.1101/2020.12.11.386789), &lt;URL:</span>
<span class="co">#&gt; https://www.biorxiv.org/content/10.1101/2020.12.11.386789v1&gt;.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; A BibTeX entry for LaTeX users is</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;   @Article{,</span>
<span class="co">#&gt;     title = {SPEAQeasy: a Scalable Pipeline for Expression Analysis and Quantification for R/Bioconductor-powered RNA-seq analyses},</span>
<span class="co">#&gt;     author = {Nicholas J. Eagles and Emily E. Burke and Jacob Leonard and Brianna K. Barry and Joshua M. Stolz and Louise Huuki and BaDoi N. Phan and Violeta Larrios Serrato and Everardo Gutiérrez-Millán and Israel Aguilar-Ordoñez and Andrew E. Jaffe and Leonardo Collado-Torres},</span>
<span class="co">#&gt;     year = {2020},</span>
<span class="co">#&gt;     journal = {bioRxiv},</span>
<span class="co">#&gt;     doi = {10.1101/2020.12.11.386789},</span>
<span class="co">#&gt;     url = {https://www.biorxiv.org/content/10.1101/2020.12.11.386789v1},</span>
<span class="co">#&gt;   }</span></code></pre></div>
</div>
</div>
<div id="workshop" class="section level1">
<h1 class="hasAnchor">
<a href="#workshop" class="anchor"></a>Workshop</h1>
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<div id="speaqeasy-overview" class="section level3">
<h3 class="hasAnchor">
<a href="#speaqeasy-overview" class="anchor"></a>SPEAQeasy Overview</h3>
<p>We introduce <a href="https://github.com/LieberInstitute/SPEAQeasy">SPEAQeasy</a>, a <strong>S</strong>calable RNA-seq processing <strong>P</strong>ipeline for <strong>E</strong>xpression <strong>A</strong>nalysis and <strong>Q</strong>uantification, that is <strong>easy</strong> to install, use, and share with others. More detailed documentation is <a href="http://research.libd.org/SPEAQeasy/">here</a>.</p>
<div class="figure">
<img src="images/workflow.png" alt=""><p class="caption">Functional workflow. Yellow coloring denotes steps which are optional or not always performed.</p>
</div>
</div>
<div id="speaqeasy-outputs" class="section level3">
<h3 class="hasAnchor">
<a href="#speaqeasy-outputs" class="anchor"></a>SPEAQeasy Outputs</h3>
<p>SPEAQeasy produces <a href="https://bioconductor.org/packages/release/bioc/html/SummarizedExperiment.html">RangedSummarizedExperiment</a> objects with raw and normalized counts for each feature type: genes, exons, exon-exon junctions, and transcripts.</p>
<p>For human samples, variant calling is performed at a list of highly variables sites. A single <a href="https://www.internationalgenome.org/wiki/Analysis/Variant%20Call%20Format/vcf-variant-call-format-version-40/">VCF file</a> is generated, containing genotype calls for all samples. This estalishes a sufficiently unique profile for each sample which can be compared against pre-sequencing genotype calls to resolve potential identity issues.</p>
<p>Optionally, <a href="https://genome.ucsc.edu/goldenPath/help/bigWig.html">BigWig</a> coverage files by strand, and averaged across strands, are produced. An additional step utilizes the <a href="https://bioconductor.org/packages/release/bioc/html/derfinder.html">derfinder</a> package to compute expressed regions, a precursor to analyses finding differentially expressed regions (DERs).</p>
<div class="figure">
<img src="images/main_outputs.png" alt=""><p class="caption">Main SPEAQeasy outputs of interest.</p>
</div>
</div>
<div id="installation-and-set-up" class="section level3">
<h3 class="hasAnchor">
<a href="#installation-and-set-up" class="anchor"></a>Installation and Set-up</h3>
<p>First, clone the SPEAQeasy repository.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1"></a><span class="fu">git</span> clone git@github.com:LieberInstitute/SPEAQeasy.git</span></code></pre></div>
<p>SPEAQeasy dependencies can be managed with docker, involving a fairly quick installation that is reproducible regardless of the computing environment. For those who are able to use docker, SPEAQeasy can be installed with:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1"></a><span class="co">#  From within the SPEAQeasy repository:</span></span>
<span id="cb5-2"><a href="#cb5-2"></a><span class="fu">bash</span> install_software.sh <span class="st">"docker"</span></span></code></pre></div>
<p>If you do not have access to docker on your machine, SPEAQeasy can be installed with:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1"></a><span class="co">#  From within the SPEAQeasy repository:</span></span>
<span id="cb6-2"><a href="#cb6-2"></a><span class="fu">bash</span> install_software.sh <span class="st">"local"</span></span></code></pre></div>
<p>Please note that installation with docker is recommended if available, and that “local” installation can take quite some time, as all dependencies are built from source. <strong>Installing and running SPEAQeasy is not required for this workshop</strong>, but is demonstrated to familiarize participants with the process. The workshop files include outputs from SPEAQeasy, which can be used later for the example differential expression analysis.</p>
<p>Choose a “main” script as appropriate for your particular set-up. “Main” scripts and associated configuration files exist for SLURM-managed computing clusters, SGE-managed clusters, local machines, and the <a href="https://jhpce.jhu.edu/">JHPCE</a> cluster.</p>
<table class="table">
<colgroup>
<col width="23%">
<col width="30%">
<col width="45%">
</colgroup>
<thead><tr class="header">
<th>Environment</th>
<th>“Main” script</th>
<th>Config file</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>SLURM cluster</td>
<td>run_pipeline_slurm.sh</td>
<td>conf/slurm.config or conf/docker_slurm.config</td>
</tr>
<tr class="even">
<td>SGE cluster</td>
<td>run_pipeline_sge.sh</td>
<td>conf/sge.config or conf/docker_sge.config</td>
</tr>
<tr class="odd">
<td>Local machine</td>
<td>run_pipeline_local.sh</td>
<td>conf/local.config or conf/docker_local.config</td>
</tr>
<tr class="even">
<td>JHPCE cluster</td>
<td>run_pipeline_jhpce.sh</td>
<td>conf/jhpce.config</td>
</tr>
</tbody>
</table>
<p>Within the main script, you can configure arguments specific to the experiment, such as the reference organism, pairing of samples, and where to place output files, among other specifications.</p>
<p>When running SPEAQeasy on a cluster (i.e. SLURM, SGE, or JHPCE users), it is recommended you submit the pipeline as a job, using the command appropriate for your cluster. For those running SPEAQeasy locally, the main script can be executed directly.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1"></a><span class="co">#  SLURM-managed clusters</span></span>
<span id="cb7-2"><a href="#cb7-2"></a><span class="ex">sbatch</span> run_pipeline_slurm.sh</span>
<span id="cb7-3"><a href="#cb7-3"></a></span>
<span id="cb7-4"><a href="#cb7-4"></a><span class="co">#  SGE-managed clusters</span></span>
<span id="cb7-5"><a href="#cb7-5"></a><span class="ex">qsub</span> run_pipeline_sge.sh</span>
<span id="cb7-6"><a href="#cb7-6"></a></span>
<span id="cb7-7"><a href="#cb7-7"></a><span class="co">#  local machines</span></span>
<span id="cb7-8"><a href="#cb7-8"></a><span class="fu">bash</span> run_pipeline_local.sh</span>
<span id="cb7-9"><a href="#cb7-9"></a></span>
<span id="cb7-10"><a href="#cb7-10"></a><span class="co">#  The JHPCE cluster</span></span>
<span id="cb7-11"><a href="#cb7-11"></a><span class="ex">qsub</span> run_pipeline_jhpce.sh</span></code></pre></div>
<p>Here’s an example of such a file.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Root for our example files</span>
<span class="va">example_files</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span>
    <span class="st">"extdata"</span>,
    <span class="st">"SPEAQeasy-example"</span>,
    package <span class="op">=</span> <span class="st">"SPEAQeasyWorkshop2020"</span>
<span class="op">)</span>

<span class="co">## Read in the actual bash script used to run SPEAQeasy with the</span>
<span class="co">## data from SPEAQeasy-example</span>
<span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/readLines.html">readLines</a></span><span class="op">(</span>
    <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">example_files</span>, <span class="st">"pipeline_outputs"</span>, <span class="st">"run_pipeline_jhpce.sh"</span><span class="op">)</span>
<span class="op">)</span>, <span class="st">"\n"</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; #!/bin/bash</span>
<span class="co">#&gt;  #$ -l bluejay,mem_free=60G,h_vmem=60G,h_fsize=800G</span>
<span class="co">#&gt;  #$ -N SPEAQeasy_ex</span>
<span class="co">#&gt;  #$ -o ./SPEAQeasy_ex.log</span>
<span class="co">#&gt;  #$ -e ./SPEAQeasy_ex.log</span>
<span class="co">#&gt;  #$ -cwd</span>
<span class="co">#&gt;  </span>
<span class="co">#&gt;  module load nextflow</span>
<span class="co">#&gt;  export _JAVA_OPTIONS="-Xms8g -Xmx10g"</span>
<span class="co">#&gt;  </span>
<span class="co">#&gt;  nextflow /dcl01/lieber/ajaffe/Nick/RNAsp/main.nf \</span>
<span class="co">#&gt;      --sample "paired" \</span>
<span class="co">#&gt;      --reference "hg38" \</span>
<span class="co">#&gt;      --strand "reverse" \</span>
<span class="co">#&gt;      --coverage \</span>
<span class="co">#&gt;      --annotation "/dcl01/lieber/ajaffe/Nick/RNAsp/Annotation" \</span>
<span class="co">#&gt;      -with-report execution_reports/JHPCE_run.html \</span>
<span class="co">#&gt;      -with-dag execution_DAGs/JHPCE_run.html \</span>
<span class="co">#&gt;      -profile jhpce \</span>
<span class="co">#&gt;      -w "/dcl01/lieber/ajaffe/lab/RNAsp_work/runs" \</span>
<span class="co">#&gt;      --input "/dcl01/lieber/ajaffe/lab/SPEAQeasy-example/sample_selection" \</span>
<span class="co">#&gt;      --output "/dcl01/lieber/ajaffe/lab/SPEAQeasy-example/pipeline_outputs"</span>
<span class="co">#&gt;  </span>
<span class="co">#&gt;  #  Produces a report for each sample tracing the pipeline steps</span>
<span class="co">#&gt;  #  performed (can be helpful for debugging).</span>
<span class="co">#&gt;  #</span>
<span class="co">#&gt;  #  Note that the reports are generated from the output log produced in the above</span>
<span class="co">#&gt;  #  section, and so if you rename the log, you must also pass replace the filename</span>
<span class="co">#&gt;  #  in the bash call below.</span>
<span class="co">#&gt;  #echo "Generating per-sample logs for debugging..."</span>
<span class="co">#&gt;  #bash /dcl01/lieber/ajaffe/Nick/RNAsp/scripts/generate_logs.sh $PWD/SPEAQeasy_output.log</span></code></pre></div>
</div>
<div id="examining-speaqeasy-outputs" class="section level3">
<h3 class="hasAnchor">
<a href="#examining-speaqeasy-outputs" class="anchor"></a>Examining SPEAQeasy outputs</h3>
<p>Let’s take a look at one of the major outputs from SPEAQeasy- the <code>RangedSummarizedExperiment</code> containing gene counts.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://LieberInstitute.github.io/SPEAQeasyWorkshop2020/">"SPEAQeasyWorkshop2020"</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://here.r-lib.org/">"here"</a></span><span class="op">)</span>
<span class="co">#&gt; here() starts at /__w/SPEAQeasyWorkshop2020/SPEAQeasyWorkshop2020</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.org/packages/SummarizedExperiment">"SummarizedExperiment"</a></span><span class="op">)</span>
<span class="co">#&gt; Loading required package: MatrixGenerics</span>
<span class="co">#&gt; Loading required package: matrixStats</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'MatrixGenerics'</span>
<span class="co">#&gt; The following objects are masked from 'package:matrixStats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,</span>
<span class="co">#&gt;     colCounts, colCummaxs, colCummins, colCumprods, colCumsums,</span>
<span class="co">#&gt;     colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,</span>
<span class="co">#&gt;     colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,</span>
<span class="co">#&gt;     colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,</span>
<span class="co">#&gt;     colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,</span>
<span class="co">#&gt;     colWeightedMeans, colWeightedMedians, colWeightedSds,</span>
<span class="co">#&gt;     colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,</span>
<span class="co">#&gt;     rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,</span>
<span class="co">#&gt;     rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,</span>
<span class="co">#&gt;     rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,</span>
<span class="co">#&gt;     rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,</span>
<span class="co">#&gt;     rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,</span>
<span class="co">#&gt;     rowWeightedMads, rowWeightedMeans, rowWeightedMedians,</span>
<span class="co">#&gt;     rowWeightedSds, rowWeightedVars</span>
<span class="co">#&gt; Loading required package: GenomicRanges</span>
<span class="co">#&gt; Loading required package: stats4</span>
<span class="co">#&gt; Loading required package: BiocGenerics</span>
<span class="co">#&gt; Loading required package: parallel</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'BiocGenerics'</span>
<span class="co">#&gt; The following objects are masked from 'package:parallel':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,</span>
<span class="co">#&gt;     clusterExport, clusterMap, parApply, parCapply, parLapply,</span>
<span class="co">#&gt;     parLapplyLB, parRapply, parSapply, parSapplyLB</span>
<span class="co">#&gt; The following objects are masked from 'package:stats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     IQR, mad, sd, var, xtabs</span>
<span class="co">#&gt; The following objects are masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     anyDuplicated, append, as.data.frame, basename, cbind, colnames,</span>
<span class="co">#&gt;     dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,</span>
<span class="co">#&gt;     grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,</span>
<span class="co">#&gt;     order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,</span>
<span class="co">#&gt;     rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,</span>
<span class="co">#&gt;     union, unique, unsplit, which.max, which.min</span>
<span class="co">#&gt; Loading required package: S4Vectors</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'S4Vectors'</span>
<span class="co">#&gt; The following object is masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     expand.grid</span>
<span class="co">#&gt; Loading required package: IRanges</span>
<span class="co">#&gt; Loading required package: GenomeInfoDb</span>
<span class="co">#&gt; Loading required package: Biobase</span>
<span class="co">#&gt; Welcome to Bioconductor</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     Vignettes contain introductory material; view with</span>
<span class="co">#&gt;     'browseVignettes()'. To cite Bioconductor, see</span>
<span class="co">#&gt;     'citation("Biobase")', and for packages 'citation("pkgname")'.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'Biobase'</span>
<span class="co">#&gt; The following object is masked from 'package:MatrixGenerics':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     rowMedians</span>
<span class="co">#&gt; The following objects are masked from 'package:matrixStats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     anyMissing, rowMedians</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/LieberInstitute/jaffelab">"jaffelab"</a></span><span class="op">)</span> <span class="co"># GitHub: LieberInstitute/jaffelab</span>
<span class="co">#&gt; Loading required package: rafalib</span>

<span class="co">#  Load the RSE containing gene counts</span>
<span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span>
    <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span>
        <span class="va">example_files</span>,
        <span class="st">"pipeline_outputs"</span>,
        <span class="st">"count_objects"</span>,
        <span class="st">"rse_gene_Jlab_experiment_n42.Rdata"</span>
    <span class="op">)</span>,
    verbose <span class="op">=</span> <span class="cn">TRUE</span>
<span class="op">)</span>
<span class="co">#&gt; Loading objects:</span>
<span class="co">#&gt;   rse_gene</span>

<span class="co">#  Print an overview</span>
<span class="va">rse_gene</span>
<span class="co">#&gt; class: RangedSummarizedExperiment </span>
<span class="co">#&gt; dim: 60609 42 </span>
<span class="co">#&gt; metadata(0):</span>
<span class="co">#&gt; assays(1): counts</span>
<span class="co">#&gt; rownames(60609): ENSG00000223972.5 ENSG00000227232.5 ...</span>
<span class="co">#&gt;   ENSG00000210195.2 ENSG00000210196.2</span>
<span class="co">#&gt; rowData names(10): Length gencodeID ... NumTx gencodeTx</span>
<span class="co">#&gt; colnames(42): R13896_H7JKMBBXX R13903_HCTYLBBXX ... R15120_HFY2MBBXX</span>
<span class="co">#&gt;   R15134_HFFGHBBXX</span>
<span class="co">#&gt; colData names(61): SAMPLE_ID FQCbasicStats ...</span>
<span class="co">#&gt;   gene_Unassigned_Ambiguity rRNA_rate</span>

<span class="co">#  Check out the metrics SPEAQeasy collected for each sample, including stats</span>
<span class="co">#  from FastQC, trimming, and alignment</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt;  [1] "SAMPLE_ID"                          "FQCbasicStats"                     </span>
<span class="co">#&gt;  [3] "perBaseQual"                        "perTileQual"                       </span>
<span class="co">#&gt;  [5] "perSeqQual"                         "perBaseContent"                    </span>
<span class="co">#&gt;  [7] "GCcontent"                          "Ncontent"                          </span>
<span class="co">#&gt;  [9] "SeqLengthDist"                      "SeqDuplication"                    </span>
<span class="co">#&gt; [11] "OverrepSeqs"                        "AdapterContent"                    </span>
<span class="co">#&gt; [13] "KmerContent"                        "SeqLength_R1"                      </span>
<span class="co">#&gt; [15] "percentGC_R1"                       "phred20-21_R1"                     </span>
<span class="co">#&gt; [17] "phred48-49_R1"                      "phred76-77_R1"                     </span>
<span class="co">#&gt; [19] "phred100-101_R1"                    "phredGT30_R1"                      </span>
<span class="co">#&gt; [21] "phredGT35_R1"                       "Adapter50-51_R1"                   </span>
<span class="co">#&gt; [23] "Adapter70-71_R1"                    "Adapter90_R1"                      </span>
<span class="co">#&gt; [25] "SeqLength_R2"                       "percentGC_R2"                      </span>
<span class="co">#&gt; [27] "phred20-21_R2"                      "phred48-49_R2"                     </span>
<span class="co">#&gt; [29] "phred76-77_R2"                      "phred100-101_R2"                   </span>
<span class="co">#&gt; [31] "phredGT30_R2"                       "phredGT35_R2"                      </span>
<span class="co">#&gt; [33] "Adapter50-51_R2"                    "Adapter70-71_R2"                   </span>
<span class="co">#&gt; [35] "Adapter90_R2"                       "bamFile"                           </span>
<span class="co">#&gt; [37] "trimmed"                            "numReads"                          </span>
<span class="co">#&gt; [39] "numMapped"                          "numUnmapped"                       </span>
<span class="co">#&gt; [41] "overallMapRate"                     "concordMapRate"                    </span>
<span class="co">#&gt; [43] "totalMapped"                        "mitoMapped"                        </span>
<span class="co">#&gt; [45] "mitoRate"                           "totalAssignedGene"                 </span>
<span class="co">#&gt; [47] "gene_Assigned"                      "gene_Unassigned_Unmapped"          </span>
<span class="co">#&gt; [49] "gene_Unassigned_Read_Type"          "gene_Unassigned_Singleton"         </span>
<span class="co">#&gt; [51] "gene_Unassigned_MappingQuality"     "gene_Unassigned_Chimera"           </span>
<span class="co">#&gt; [53] "gene_Unassigned_FragmentLength"     "gene_Unassigned_Duplicate"         </span>
<span class="co">#&gt; [55] "gene_Unassigned_MultiMapping"       "gene_Unassigned_Secondary"         </span>
<span class="co">#&gt; [57] "gene_Unassigned_NonSplit"           "gene_Unassigned_NoFeatures"        </span>
<span class="co">#&gt; [59] "gene_Unassigned_Overlapping_Length" "gene_Unassigned_Ambiguity"         </span>
<span class="co">#&gt; [61] "rRNA_rate"</span>

<span class="co">#  View the "top left corner" of the counts assay, containing counts for a few</span>
<span class="co">#  genes and a few samples</span>
<span class="fu"><a href="https://rdrr.io/pkg/jaffelab/man/corner.html">corner</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">assays</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span><span class="op">$</span><span class="va">counts</span><span class="op">)</span>
<span class="co">#&gt;                   R13896_H7JKMBBXX R13903_HCTYLBBXX R13904_H7K5NBBXX</span>
<span class="co">#&gt; ENSG00000223972.5                0                0                0</span>
<span class="co">#&gt; ENSG00000227232.5               46               36               20</span>
<span class="co">#&gt; ENSG00000278267.1                8                3                0</span>
<span class="co">#&gt; ENSG00000243485.5                0                0                0</span>
<span class="co">#&gt; ENSG00000284332.1                0                0                0</span>
<span class="co">#&gt; ENSG00000237613.2                0                0                0</span>
<span class="co">#&gt;                   R13983_H7JKMBBXX R13985_H7JM5BBXX R13986_H7JM5BBXX</span>
<span class="co">#&gt; ENSG00000223972.5                0                0                0</span>
<span class="co">#&gt; ENSG00000227232.5              118               61               47</span>
<span class="co">#&gt; ENSG00000278267.1               13                4                8</span>
<span class="co">#&gt; ENSG00000243485.5                0                1                0</span>
<span class="co">#&gt; ENSG00000284332.1                0                0                0</span>
<span class="co">#&gt; ENSG00000237613.2                0                0                0</span>

<span class="co">#  Examine what data is provided in the RSE object for each gene ("row")</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">rowData</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; DataFrame with 6 rows and 10 columns</span>
<span class="co">#&gt;                      Length         gencodeID       ensemblID</span>
<span class="co">#&gt;                   &lt;integer&gt;       &lt;character&gt;     &lt;character&gt;</span>
<span class="co">#&gt; ENSG00000223972.5      1735 ENSG00000223972.5 ENSG00000223972</span>
<span class="co">#&gt; ENSG00000227232.5      1351 ENSG00000227232.5 ENSG00000227232</span>
<span class="co">#&gt; ENSG00000278267.1        68 ENSG00000278267.1 ENSG00000278267</span>
<span class="co">#&gt; ENSG00000243485.5      1021 ENSG00000243485.5 ENSG00000243485</span>
<span class="co">#&gt; ENSG00000284332.1       138 ENSG00000284332.1 ENSG00000284332</span>
<span class="co">#&gt; ENSG00000237613.2      1219 ENSG00000237613.2 ENSG00000237613</span>
<span class="co">#&gt;                                gene_type      Symbol  EntrezID       Class</span>
<span class="co">#&gt;                              &lt;character&gt; &lt;character&gt; &lt;integer&gt; &lt;character&gt;</span>
<span class="co">#&gt; ENSG00000223972.5 transcribed_unproces..     DDX11L1        NA       InGen</span>
<span class="co">#&gt; ENSG00000227232.5 unprocessed_pseudogene      WASH7P        NA       InGen</span>
<span class="co">#&gt; ENSG00000278267.1                  miRNA   MIR6859-1        NA       InGen</span>
<span class="co">#&gt; ENSG00000243485.5                 lncRNA MIR1302-2HG        NA       InGen</span>
<span class="co">#&gt; ENSG00000284332.1                  miRNA   MIR1302-2        NA       InGen</span>
<span class="co">#&gt; ENSG00000237613.2                 lncRNA     FAM138A        NA       InGen</span>
<span class="co">#&gt;                    meanExprs     NumTx              gencodeTx</span>
<span class="co">#&gt;                    &lt;numeric&gt; &lt;integer&gt;            &lt;character&gt;</span>
<span class="co">#&gt; ENSG00000223972.5 0.00347263         2 ENST00000456328.2;EN..</span>
<span class="co">#&gt; ENSG00000227232.5 1.57167877         1      ENST00000488147.1</span>
<span class="co">#&gt; ENSG00000278267.1 7.74437941         1      ENST00000619216.1</span>
<span class="co">#&gt; ENSG00000243485.5 0.01403484         2 ENST00000473358.1;EN..</span>
<span class="co">#&gt; ENSG00000284332.1 0.00000000         1      ENST00000607096.1</span>
<span class="co">#&gt; ENSG00000237613.2 0.00683033         2 ENST00000417324.1;EN..</span></code></pre></div>
</div>
</div>
<div id="de-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#de-analysis" class="anchor"></a>DE Analysis</h2>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/Bioconductor/BiocStyle">"BiocStyle"</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://yulab-smu.top/biomedical-knowledge-mining-book/">"clusterProfiler"</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="http://bioinf.wehi.edu.au/edgeR">"edgeR"</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="http://bioinf.wehi.edu.au/limma">"limma"</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st">"pheatmap"</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st">"org.Hs.eg.db"</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st">"RColorBrewer"</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/leekgroup/recount">"recount"</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/r-lib/sessioninfo#readme">"sessioninfo"</a></span><span class="op">)</span></code></pre></div>
<p>Edit this as you see fit =)</p>
<p>Here is an example of you can cite your package inside the vignette:</p>
<ul>
<li>
<em><a href="https://bioconductor.org/packages/3.12/SPEAQeasyWorkshop2020">SPEAQeasyWorkshop2020</a></em> (Eagles, Stolz, and Collado-Torres, 2020)</li>
</ul>
</div>
</div>
<div id="acknowledgements" class="section level1">
<h1 class="hasAnchor">
<a href="#acknowledgements" class="anchor"></a>Acknowledgements</h1>
<div id="members-of-the-former-jaffe-lab" class="section level2">
<h2 class="hasAnchor">
<a href="#members-of-the-former-jaffe-lab" class="anchor"></a>Members of the (former) Jaffe Lab</h2>
<ul>
<li>Emily Burke: much of the code for SPEAQeasy before it was re-built with Nextflow, including a great portion of the R scripts still used now</li>
<li>Brianna Barry: guidance, naming, and testing SPEAQeasy, leading to features such as per-sample logs and automatic strand inference, among others</li>
<li>Louise Huuki: help with our guide involving resolving sample swaps from SPEAQeasy outputs</li>
<li>BaDoi Phan: development of the pipeline before using Nextflow</li>
<li>Andrew Jaffe: principal investigator, leading and overseeing the SPEAQeasy project</li>
<li>Leonardo Collado Torres: development, guidance, and help throughout the SPEAQeasy project; development of this workshop package and website</li>
</ul>
</div>
<div id="winter-genomics-team" class="section level2">
<h2 class="hasAnchor">
<a href="#winter-genomics-team" class="anchor"></a>Winter Genomics Team</h2>
<p>Estlabished a foundation for the Nextflow-port (now SPEAQeasy) of the RNA-seq pipeline internally used by the Jaffe Lab.</p>
<ul>
<li>Jacob Leonard</li>
<li>Violeta Larios Serrato</li>
</ul>
</div>
</div>
<div id="reproducibility" class="section level1">
<h1 class="hasAnchor">
<a href="#reproducibility" class="anchor"></a>Reproducibility</h1>
<p>The <em><a href="https://bioconductor.org/packages/3.12/SPEAQeasyWorkshop2020">SPEAQeasyWorkshop2020</a></em> package (Eagles, Stolz, and Collado-Torres, 2020) was made possible thanks to:</p>
<ul>
<li>R (R Core Team, 2020)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.12/BiocStyle">BiocStyle</a></em> (Oleś, Morgan, and Huber, 2020)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.12/clusterProfiler">clusterProfiler</a></em> (Yu, Wang, Han, and He, 2012)</li>
<li><em><a href="https://bioconductor.org/packages/3.12/edgeR">edgeR</a></em></li>
<li>
<em><a href="https://CRAN.R-project.org/package=here">here</a></em> (Müller, 2020)</li>
<li>
<em><a href="https://github.com/LieberInstitute/jaffelab">jaffelab</a></em> (Collado-Torres, Jaffe, and Burke, 2019)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.12/limma">limma</a></em> (Ritchie, Phipson, Wu, Hu, Law, Shi, and Smyth, 2015)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=knitr">knitr</a></em> (Xie, 2020)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.12/org.Hs.eg.db">org.Hs.eg.db</a></em> (Carlson, 2020)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=pheatmap">pheatmap</a></em> (Kolde, 2019)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=RColorBrewer">RColorBrewer</a></em> (Neuwirth, 2014)</li>
<li><em><a href="https://bioconductor.org/packages/3.12/recount">recount</a></em></li>
<li>
<em><a href="https://CRAN.R-project.org/package=RefManageR">RefManageR</a></em> (McLean, 2017)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=rmarkdown">rmarkdown</a></em> (Allaire, Xie, McPherson, Luraschi, Ushey, Atkins, Wickham, Cheng, Chang, and Iannone, 2020)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=sessioninfo">sessioninfo</a></em> (Csárdi, core, Wickham, Chang, Flight, Müller, and Hester, 2018)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.12/SummarizedExperiment">SummarizedExperiment</a></em> (Morgan, Obenchain, Hester, and Pagès, 2020)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.12/voom">voom</a></em> (Law, Chen, Shi, and Smyth, 2014)</li>
</ul>
<p>This package was developed using <em><a href="https://github.com/lcolladotor/biocthis">biocthis</a></em>.</p>
<p>Code for creating the vignette</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Create the vignette</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/rstudio/rmarkdown">"rmarkdown"</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/system.time.html">system.time</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/rmarkdown/man/render.html">render</a></span><span class="op">(</span><span class="st">"SPEAQeasyWorkshop2020.Rmd"</span>, <span class="st">"BiocStyle::html_document"</span><span class="op">)</span><span class="op">)</span>

<span class="co">## Extract the R code</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://yihui.org/knitr/">"knitr"</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/knitr/man/knit.html">knit</a></span><span class="op">(</span><span class="st">"SPEAQeasyWorkshop2020.Rmd"</span>, tangle <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p>Date the vignette was generated.</p>
<pre><code>#&gt; [1] "2020-12-15 16:33:52 UTC"</code></pre>
<p>Wallclock time spent generating the vignette.</p>
<pre><code>#&gt; Time difference of 17.08 secs</code></pre>
<p><code>R</code> session information.</p>
<pre><code>#&gt; ─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────
#&gt;  setting  value                       
#&gt;  version  R version 4.0.3 (2020-10-10)
#&gt;  os       Ubuntu 20.04 LTS            
#&gt;  system   x86_64, linux-gnu           
#&gt;  ui       X11                         
#&gt;  language (EN)                        
#&gt;  collate  en_US.UTF-8                 
#&gt;  ctype    en_US.UTF-8                 
#&gt;  tz       Etc/UTC                     
#&gt;  date     2020-12-15                  
#&gt; 
#&gt; ─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────
#&gt;  package               * version  date       lib source                                   
#&gt;  AnnotationDbi         * 1.52.0   2020-10-27 [1] Bioconductor                             
#&gt;  askpass                 1.1      2019-01-13 [2] RSPM (R 4.0.0)                           
#&gt;  assertthat              0.2.1    2019-03-21 [2] RSPM (R 4.0.0)                           
#&gt;  backports               1.2.1    2020-12-09 [1] RSPM (R 4.0.3)                           
#&gt;  base64enc               0.1-3    2015-07-28 [2] RSPM (R 4.0.0)                           
#&gt;  Biobase               * 2.50.0   2020-10-27 [1] Bioconductor                             
#&gt;  BiocFileCache           1.14.0   2020-10-27 [1] Bioconductor                             
#&gt;  BiocGenerics          * 0.36.0   2020-10-27 [1] Bioconductor                             
#&gt;  BiocManager             1.30.10  2019-11-16 [2] CRAN (R 4.0.3)                           
#&gt;  BiocParallel            1.24.1   2020-11-06 [1] Bioconductor                             
#&gt;  BiocStyle             * 2.18.1   2020-11-24 [1] Bioconductor                             
#&gt;  biomaRt                 2.46.0   2020-10-27 [1] Bioconductor                             
#&gt;  Biostrings              2.58.0   2020-10-27 [1] Bioconductor                             
#&gt;  bit                     4.0.4    2020-08-04 [1] RSPM (R 4.0.2)                           
#&gt;  bit64                   4.0.5    2020-08-30 [1] RSPM (R 4.0.2)                           
#&gt;  bitops                  1.0-6    2013-08-17 [1] RSPM (R 4.0.0)                           
#&gt;  blob                    1.2.1    2020-01-20 [1] RSPM (R 4.0.0)                           
#&gt;  bookdown                0.21     2020-10-13 [1] RSPM (R 4.0.2)                           
#&gt;  BSgenome                1.58.0   2020-10-27 [1] Bioconductor                             
#&gt;  bumphunter              1.32.0   2020-10-27 [1] Bioconductor                             
#&gt;  checkmate               2.0.0    2020-02-06 [1] RSPM (R 4.0.0)                           
#&gt;  cli                     2.2.0    2020-11-20 [2] RSPM (R 4.0.3)                           
#&gt;  cluster                 2.1.0    2019-06-19 [3] CRAN (R 4.0.3)                           
#&gt;  clusterProfiler       * 3.18.0   2020-10-27 [1] Bioconductor                             
#&gt;  codetools               0.2-18   2020-11-04 [3] RSPM (R 4.0.3)                           
#&gt;  colorspace              2.0-0    2020-11-11 [1] RSPM (R 4.0.3)                           
#&gt;  cowplot                 1.1.0    2020-09-08 [1] RSPM (R 4.0.2)                           
#&gt;  crayon                  1.3.4    2017-09-16 [2] RSPM (R 4.0.0)                           
#&gt;  curl                    4.3      2019-12-02 [2] RSPM (R 4.0.0)                           
#&gt;  data.table              1.13.4   2020-12-08 [1] RSPM (R 4.0.3)                           
#&gt;  DBI                     1.1.0    2019-12-15 [1] RSPM (R 4.0.0)                           
#&gt;  dbplyr                  2.0.0    2020-11-03 [1] RSPM (R 4.0.3)                           
#&gt;  DelayedArray            0.16.0   2020-10-27 [1] Bioconductor                             
#&gt;  derfinder               1.24.0   2020-10-27 [1] Bioconductor                             
#&gt;  derfinderHelper         1.24.0   2020-10-27 [1] Bioconductor                             
#&gt;  desc                    1.2.0    2018-05-01 [2] RSPM (R 4.0.0)                           
#&gt;  digest                  0.6.27   2020-10-24 [2] RSPM (R 4.0.3)                           
#&gt;  DO.db                   2.9      2020-11-25 [1] Bioconductor                             
#&gt;  doRNG                   1.8.2    2020-01-27 [1] RSPM (R 4.0.0)                           
#&gt;  DOSE                    3.16.0   2020-10-27 [1] Bioconductor                             
#&gt;  downloader              0.4      2015-07-09 [1] RSPM (R 4.0.0)                           
#&gt;  dplyr                   1.0.2    2020-08-18 [1] RSPM (R 4.0.2)                           
#&gt;  edgeR                 * 3.32.0   2020-10-27 [1] Bioconductor                             
#&gt;  ellipsis                0.3.1    2020-05-15 [2] RSPM (R 4.0.0)                           
#&gt;  enrichplot              1.10.1   2020-11-14 [1] Bioconductor                             
#&gt;  evaluate                0.14     2019-05-28 [2] RSPM (R 4.0.0)                           
#&gt;  fansi                   0.4.1    2020-01-08 [2] RSPM (R 4.0.0)                           
#&gt;  farver                  2.0.3    2020-01-16 [1] RSPM (R 4.0.0)                           
#&gt;  fastmatch               1.1-0    2017-01-28 [1] RSPM (R 4.0.0)                           
#&gt;  fgsea                   1.16.0   2020-10-27 [1] Bioconductor                             
#&gt;  foreach                 1.5.1    2020-10-15 [1] RSPM (R 4.0.2)                           
#&gt;  foreign                 0.8-80   2020-05-24 [3] CRAN (R 4.0.3)                           
#&gt;  Formula                 1.2-4    2020-10-16 [1] RSPM (R 4.0.2)                           
#&gt;  fs                      1.5.0    2020-07-31 [2] RSPM (R 4.0.2)                           
#&gt;  generics                0.1.0    2020-10-31 [1] RSPM (R 4.0.3)                           
#&gt;  GenomeInfoDb          * 1.26.2   2020-12-08 [1] Bioconductor                             
#&gt;  GenomeInfoDbData        1.2.4    2020-11-25 [1] Bioconductor                             
#&gt;  GenomicAlignments       1.26.0   2020-10-27 [1] Bioconductor                             
#&gt;  GenomicFeatures         1.42.1   2020-11-12 [1] Bioconductor                             
#&gt;  GenomicFiles            1.26.0   2020-10-27 [1] Bioconductor                             
#&gt;  GenomicRanges         * 1.42.0   2020-10-27 [1] Bioconductor                             
#&gt;  GEOquery                2.58.0   2020-10-27 [1] Bioconductor                             
#&gt;  ggforce                 0.3.2    2020-06-23 [1] RSPM (R 4.0.2)                           
#&gt;  ggplot2                 3.3.2    2020-06-19 [1] RSPM (R 4.0.1)                           
#&gt;  ggraph                  2.0.4    2020-11-16 [1] RSPM (R 4.0.3)                           
#&gt;  ggrepel                 0.8.2    2020-03-08 [1] RSPM (R 4.0.2)                           
#&gt;  glue                    1.4.2    2020-08-27 [2] RSPM (R 4.0.2)                           
#&gt;  GO.db                   3.12.1   2020-11-25 [1] Bioconductor                             
#&gt;  googledrive             1.0.1    2020-05-05 [1] RSPM (R 4.0.0)                           
#&gt;  GOSemSim                2.16.1   2020-10-29 [1] Bioconductor                             
#&gt;  graphlayouts            0.7.1    2020-10-26 [1] RSPM (R 4.0.3)                           
#&gt;  gridExtra               2.3      2017-09-09 [1] RSPM (R 4.0.0)                           
#&gt;  gtable                  0.3.0    2019-03-25 [1] RSPM (R 4.0.0)                           
#&gt;  here                  * 1.0.1    2020-12-13 [1] RSPM (R 4.0.3)                           
#&gt;  Hmisc                   4.4-2    2020-11-29 [1] RSPM (R 4.0.3)                           
#&gt;  hms                     0.5.3    2020-01-08 [1] RSPM (R 4.0.0)                           
#&gt;  htmlTable               2.1.0    2020-09-16 [1] RSPM (R 4.0.2)                           
#&gt;  htmltools               0.5.0    2020-06-16 [2] RSPM (R 4.0.1)                           
#&gt;  htmlwidgets             1.5.3    2020-12-10 [2] RSPM (R 4.0.3)                           
#&gt;  httr                    1.4.2    2020-07-20 [2] RSPM (R 4.0.2)                           
#&gt;  igraph                  1.2.6    2020-10-06 [1] RSPM (R 4.0.2)                           
#&gt;  IRanges               * 2.24.1   2020-12-12 [1] Bioconductor                             
#&gt;  iterators               1.0.13   2020-10-15 [1] RSPM (R 4.0.2)                           
#&gt;  jaffelab              * 0.99.30  2020-11-25 [1] Github (LieberInstitute/jaffelab@42637ff)
#&gt;  jpeg                    0.1-8.1  2019-10-24 [1] RSPM (R 4.0.0)                           
#&gt;  jsonlite                1.7.2    2020-12-09 [2] RSPM (R 4.0.3)                           
#&gt;  knitr                   1.30     2020-09-22 [2] RSPM (R 4.0.2)                           
#&gt;  lattice                 0.20-41  2020-04-02 [3] CRAN (R 4.0.3)                           
#&gt;  latticeExtra            0.6-29   2019-12-19 [1] RSPM (R 4.0.0)                           
#&gt;  lifecycle               0.2.0    2020-03-06 [2] RSPM (R 4.0.0)                           
#&gt;  limma                 * 3.46.0   2020-10-27 [1] Bioconductor                             
#&gt;  locfit                  1.5-9.4  2020-03-25 [1] RSPM (R 4.0.0)                           
#&gt;  lubridate               1.7.9.2  2020-11-13 [1] RSPM (R 4.0.3)                           
#&gt;  magrittr                2.0.1    2020-11-17 [2] RSPM (R 4.0.3)                           
#&gt;  MASS                    7.3-53   2020-09-09 [3] CRAN (R 4.0.3)                           
#&gt;  Matrix                  1.2-18   2019-11-27 [3] CRAN (R 4.0.3)                           
#&gt;  MatrixGenerics        * 1.2.0    2020-10-27 [1] Bioconductor                             
#&gt;  matrixStats           * 0.57.0   2020-09-25 [1] RSPM (R 4.0.2)                           
#&gt;  memoise                 1.1.0    2017-04-21 [2] RSPM (R 4.0.0)                           
#&gt;  munsell                 0.5.0    2018-06-12 [1] RSPM (R 4.0.0)                           
#&gt;  nnet                    7.3-14   2020-04-26 [3] CRAN (R 4.0.3)                           
#&gt;  openssl                 1.4.3    2020-09-18 [2] RSPM (R 4.0.2)                           
#&gt;  org.Hs.eg.db          * 3.12.0   2020-11-25 [1] Bioconductor                             
#&gt;  pheatmap              * 1.0.12   2019-01-04 [1] RSPM (R 4.0.0)                           
#&gt;  pillar                  1.4.7    2020-11-20 [2] RSPM (R 4.0.3)                           
#&gt;  pkgconfig               2.0.3    2019-09-22 [2] RSPM (R 4.0.0)                           
#&gt;  pkgdown                 1.6.1    2020-09-12 [1] RSPM (R 4.0.2)                           
#&gt;  plyr                    1.8.6    2020-03-03 [1] RSPM (R 4.0.2)                           
#&gt;  png                     0.1-7    2013-12-03 [1] RSPM (R 4.0.0)                           
#&gt;  polyclip                1.10-0   2019-03-14 [1] RSPM (R 4.0.0)                           
#&gt;  prettyunits             1.1.1    2020-01-24 [2] RSPM (R 4.0.0)                           
#&gt;  progress                1.2.2    2019-05-16 [1] RSPM (R 4.0.0)                           
#&gt;  purrr                   0.3.4    2020-04-17 [2] RSPM (R 4.0.0)                           
#&gt;  qvalue                  2.22.0   2020-10-27 [1] Bioconductor                             
#&gt;  R6                      2.5.0    2020-10-28 [2] RSPM (R 4.0.3)                           
#&gt;  rafalib               * 1.0.0    2015-08-09 [1] RSPM (R 4.0.0)                           
#&gt;  ragg                    0.4.0    2020-10-05 [1] RSPM (R 4.0.3)                           
#&gt;  rappdirs                0.3.1    2016-03-28 [1] RSPM (R 4.0.0)                           
#&gt;  RColorBrewer          * 1.1-2    2014-12-07 [1] RSPM (R 4.0.0)                           
#&gt;  Rcpp                    1.0.5    2020-07-06 [2] RSPM (R 4.0.2)                           
#&gt;  RCurl                   1.98-1.2 2020-04-18 [1] RSPM (R 4.0.0)                           
#&gt;  readr                   1.4.0    2020-10-05 [1] RSPM (R 4.0.3)                           
#&gt;  recount               * 1.16.0   2020-10-27 [1] Bioconductor                             
#&gt;  RefManageR            * 1.3.0    2020-11-25 [1] Github (ropensci/RefManageR@ab8fe60)     
#&gt;  rentrez                 1.2.3    2020-11-10 [1] RSPM (R 4.0.3)                           
#&gt;  reshape2                1.4.4    2020-04-09 [1] RSPM (R 4.0.2)                           
#&gt;  rlang                   0.4.9    2020-11-26 [2] RSPM (R 4.0.3)                           
#&gt;  rmarkdown               2.5      2020-10-21 [1] RSPM (R 4.0.3)                           
#&gt;  rngtools                1.5      2020-01-23 [1] RSPM (R 4.0.0)                           
#&gt;  rpart                   4.1-15   2019-04-12 [3] CRAN (R 4.0.3)                           
#&gt;  rprojroot               2.0.2    2020-11-15 [2] RSPM (R 4.0.3)                           
#&gt;  Rsamtools               2.6.0    2020-10-27 [1] Bioconductor                             
#&gt;  RSQLite                 2.2.1    2020-09-30 [1] RSPM (R 4.0.2)                           
#&gt;  rstudioapi              0.13     2020-11-12 [2] RSPM (R 4.0.3)                           
#&gt;  rtracklayer             1.50.0   2020-10-27 [1] Bioconductor                             
#&gt;  rvcheck                 0.1.8    2020-03-01 [1] RSPM (R 4.0.0)                           
#&gt;  S4Vectors             * 0.28.1   2020-12-09 [1] Bioconductor                             
#&gt;  scales                  1.1.1    2020-05-11 [1] RSPM (R 4.0.0)                           
#&gt;  scatterpie              0.1.5    2020-09-09 [1] RSPM (R 4.0.2)                           
#&gt;  segmented               1.3-1    2020-12-10 [1] RSPM (R 4.0.3)                           
#&gt;  sessioninfo           * 1.1.1    2018-11-05 [2] RSPM (R 4.0.0)                           
#&gt;  shadowtext              0.0.7    2019-11-06 [1] RSPM (R 4.0.0)                           
#&gt;  SPEAQeasyWorkshop2020 * 0.99.0   2020-12-15 [1] Bioconductor                             
#&gt;  stringi                 1.5.3    2020-09-09 [2] RSPM (R 4.0.2)                           
#&gt;  stringr                 1.4.0    2019-02-10 [2] RSPM (R 4.0.0)                           
#&gt;  SummarizedExperiment  * 1.20.0   2020-10-27 [1] Bioconductor                             
#&gt;  survival                3.2-7    2020-09-28 [3] CRAN (R 4.0.3)                           
#&gt;  systemfonts             0.3.2    2020-09-29 [1] RSPM (R 4.0.3)                           
#&gt;  textshaping             0.2.1    2020-11-13 [1] RSPM (R 4.0.3)                           
#&gt;  tibble                  3.0.4    2020-10-12 [2] RSPM (R 4.0.2)                           
#&gt;  tidygraph               1.2.0    2020-05-12 [1] RSPM (R 4.0.2)                           
#&gt;  tidyr                   1.1.2    2020-08-27 [1] RSPM (R 4.0.3)                           
#&gt;  tidyselect              1.1.0    2020-05-11 [1] RSPM (R 4.0.0)                           
#&gt;  tweenr                  1.0.1    2018-12-14 [1] RSPM (R 4.0.2)                           
#&gt;  VariantAnnotation       1.36.0   2020-10-27 [1] Bioconductor                             
#&gt;  vctrs                   0.3.5    2020-11-17 [2] RSPM (R 4.0.3)                           
#&gt;  viridis                 0.5.1    2018-03-29 [1] RSPM (R 4.0.0)                           
#&gt;  viridisLite             0.3.0    2018-02-01 [1] RSPM (R 4.0.0)                           
#&gt;  withr                   2.3.0    2020-09-22 [2] RSPM (R 4.0.2)                           
#&gt;  xfun                    0.19     2020-10-30 [2] RSPM (R 4.0.3)                           
#&gt;  XML                     3.99-0.5 2020-07-23 [1] RSPM (R 4.0.2)                           
#&gt;  xml2                    1.3.2    2020-04-23 [2] RSPM (R 4.0.0)                           
#&gt;  XVector                 0.30.0   2020-10-27 [1] Bioconductor                             
#&gt;  yaml                    2.2.1    2020-02-01 [2] RSPM (R 4.0.0)                           
#&gt;  zlibbioc                1.36.0   2020-10-27 [1] Bioconductor                             
#&gt; 
#&gt; [1] /__w/_temp/Library
#&gt; [2] /usr/local/lib/R/site-library
#&gt; [3] /usr/local/lib/R/library</code></pre>
</div>
<div id="bibliography" class="section level1">
<h1 class="hasAnchor">
<a href="#bibliography" class="anchor"></a>Bibliography</h1>
<p>This vignette was generated using <em><a href="https://bioconductor.org/packages/3.12/BiocStyle">BiocStyle</a></em> (Oleś, Morgan, and Huber, 2020) with <em><a href="https://CRAN.R-project.org/package=knitr">knitr</a></em> (Xie, 2020) and <em><a href="https://CRAN.R-project.org/package=rmarkdown">rmarkdown</a></em> (Allaire, Xie, McPherson, et al., 2020) running behind the scenes.</p>
<p>Citations made with <em><a href="https://CRAN.R-project.org/package=RefManageR">RefManageR</a></em> (McLean, 2017).</p>
<p>
<a id="bib-allaire2020rmarkdown"></a><a href="#cite-allaire2020rmarkdown">[1]</a><cite> J. Allaire, Y. Xie, J. McPherson, et al. <em>rmarkdown: Dynamic Documents for R</em>. R package version 2.5. 2020. URL: <a href="https://github.com/rstudio/rmarkdown">https://github.com/rstudio/rmarkdown</a>.</cite>
</p>
<p>
<a id="bib-carlson2020genome"></a><a href="#cite-carlson2020genome">[2]</a><cite> M. Carlson. <em>org.Hs.eg.db: Genome wide annotation for Human</em>. R package version 3.12.0. 2020.</cite>
</p>
<p>
<a id="bib-colladotorres2019jaffelab"></a><a href="#cite-colladotorres2019jaffelab">[3]</a><cite> L. Collado-Torres, A. E. Jaffe, and E. E. Burke. <em>jaffelab: Commonly used functions by the Jaffe lab</em>. R package version 0.99.30. 2019. URL: <a href="https://github.com/LieberInstitute/jaffelab">https://github.com/LieberInstitute/jaffelab</a>.</cite>
</p>
<p>
<a id="bib-csrdi2018sessioninfo"></a><a href="#cite-csrdi2018sessioninfo">[4]</a><cite> G. Csárdi, R. core, H. Wickham, et al. <em>sessioninfo: R Session Information</em>. R package version 1.1.1. 2018. URL: <a href="https://github.com/r-lib/sessioninfo#readme">https://github.com/r-lib/sessioninfo#readme</a>.</cite>
</p>
<p>
<a id="bib-eagles2020speaqeasy"></a><a href="#cite-eagles2020speaqeasy">[5]</a><cite> N. J. Eagles, E. E. Burke, J. Leonard, et al. “SPEAQeasy: a Scalable Pipeline for Expression Analysis and Quantification for R/Bioconductor-powered RNA-seq analyses”. In: <em>bioRxiv</em> (2020). DOI: <a href="https://doi.org/10.1101/2020.12.11.386789">10.1101/2020.12.11.386789</a>. URL: <a href="https://www.biorxiv.org/content/10.1101/2020.12.11.386789v1">https://www.biorxiv.org/content/10.1101/2020.12.11.386789v1</a>.</cite>
</p>
<p>
<a id="bib-eagles2020eurobioc2020"></a><a href="#cite-eagles2020eurobioc2020">[6]</a><cite> N. J. Eagles, J. M. Stolz, and L. Collado-Torres. <em>EuroBioc2020 SPEAQeasy workshop</em>. <a href="https://github.com/LieberInstitute/SPEAQeasyWorkshop2020" class="uri">https://github.com/LieberInstitute/SPEAQeasyWorkshop2020</a> - R package version 0.99.0. 2020. DOI: <a href="https://doi.org/10.18129/B9.bioc.SPEAQeasyWorkshop2020">10.18129/B9.bioc.SPEAQeasyWorkshop2020</a>. URL: <a href="http://www.bioconductor.org/packages/SPEAQeasyWorkshop2020">http://www.bioconductor.org/packages/SPEAQeasyWorkshop2020</a>.</cite>
</p>
<p>
<a id="bib-kolde2019pheatmap"></a><a href="#cite-kolde2019pheatmap">[7]</a><cite> R. Kolde. <em>pheatmap: Pretty Heatmaps</em>. R package version 1.0.12. 2019.</cite>
</p>
<p>
<a id="bib-voom"></a><a href="#cite-voom">[8]</a><cite> C. Law, Y. Chen, W. Shi, et al. “Voom: precision weights unlock linear model analysis tools for RNA-seq read counts”. In: <em>Genome Biology</em> 15 (2014), p. R29.</cite>
</p>
<p>
<a id="bib-mclean2017refmanager"></a><a href="#cite-mclean2017refmanager">[9]</a><cite> M. W. McLean. “RefManageR: Import and Manage BibTeX and BibLaTeX References in R”. In: <em>The Journal of Open Source Software</em> (2017). DOI: <a href="https://doi.org/10.21105/joss.00338">10.21105/joss.00338</a>.</cite>
</p>
<p>
<a id="bib-morgan2020summarizedexperiment"></a><a href="#cite-morgan2020summarizedexperiment">[10]</a><cite> M. Morgan, V. Obenchain, J. Hester, et al. <em>SummarizedExperiment: SummarizedExperiment container</em>. R package version 1.20.0. 2020. URL: <a href="https://bioconductor.org/packages/SummarizedExperiment">https://bioconductor.org/packages/SummarizedExperiment</a>.</cite>
</p>
<p>
<a id="bib-mller2020here"></a><a href="#cite-mller2020here">[11]</a><cite> K. Müller. <em>here: A Simpler Way to Find Your Files</em>. <a href="https://here.r-lib.org/" class="uri">https://here.r-lib.org/</a>, <a href="https://github.com/r-lib/here" class="uri">https://github.com/r-lib/here</a>. 2020.</cite>
</p>
<p>
<a id="bib-neuwirth2014rcolorbrewer"></a><a href="#cite-neuwirth2014rcolorbrewer">[12]</a><cite> E. Neuwirth. <em>RColorBrewer: ColorBrewer Palettes</em>. R package version 1.1-2. 2014.</cite>
</p>
<p>
<a id="bib-ole2020biocstyle"></a><a href="#cite-ole2020biocstyle">[13]</a><cite> A. Oleś, M. Morgan, and W. Huber. <em>BiocStyle: Standard styles for vignettes and other Bioconductor documents</em>. R package version 2.18.1. 2020. URL: <a href="https://github.com/Bioconductor/BiocStyle">https://github.com/Bioconductor/BiocStyle</a>.</cite>
</p>
<p>
<a id="bib-2020language"></a><a href="#cite-2020language">[14]</a><cite> R Core Team. <em>R: A Language and Environment for Statistical Computing</em>. R Foundation for Statistical Computing. Vienna, Austria, 2020. URL: <a href="https://www.R-project.org/">https://www.R-project.org/</a>.</cite>
</p>
<p>
<a id="bib-ritchie2015limma"></a><a href="#cite-ritchie2015limma">[15]</a><cite> M. E. Ritchie, B. Phipson, D. Wu, et al. “limma powers differential expression analyses for RNA-sequencing and microarray studies”. In: <em>Nucleic Acids Research</em> 43.7 (2015), p. e47. DOI: <a href="https://doi.org/10.1093/nar/gkv007">10.1093/nar/gkv007</a>.</cite>
</p>
<p>
<a id="bib-xie2020knitr"></a><a href="#cite-xie2020knitr">[16]</a><cite> Y. Xie. <em>knitr: A General-Purpose Package for Dynamic Report Generation in R</em>. R package version 1.30. 2020. URL: <a href="https://yihui.org/knitr/">https://yihui.org/knitr/</a>.</cite>
</p>
<p>
<a id="bib-yu2012clusterprofiler"></a><a href="#cite-yu2012clusterprofiler">[17]</a><cite> G. Yu, L. Wang, Y. Han, et al. “clusterProfiler: an R package for comparing biological themes among gene clusters”. In: <em>OMICS: A Journal of Integrative Biology</em> 16.5 (2012), pp. 284-287. DOI: <a href="https://doi.org/10.1089/omi.2011.0118">10.1089/omi.2011.0118</a>.</cite>
</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Nicholas J. Eagles, Joshua M. Stolz.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
